### PR TITLE
Feature:  Add Support for Event Invitees and Registrants

### LINKED ISSUE #1644: Add Support for Event Invitees and Registrants
**Is your feature request related to a problem? Please describe.**

We need the ability to track users who:

1. Are invited to events
2. Register for events

This is important for venue management where there may be limited seating and registration is required.

**Describe the solution you'd like**

We need `[User!]` type fields added to the Event collection with the following names:

1. `invitedUsers`
2. `registeredUsers`

Other:

1. You will be required to write tests for the expected CRUD activities related to these new users being tracked.
1. The tests must include verification of the users added to the following lists:
    1. `attendees` (This testing may not have been done)
    1. `invitedUsers`
    1. `registeredUsers`

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- See above

**Additional context**

The current screen shot of the `develop` branch's schema

image

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**

- Support added for event invitees and registrants.
- Admins can invite, register, and check-in users.
- Invited users can register for events.

**Issue Number:**

Fixes #1644 

**Did you add tests for your changes?**

- Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

- The eventAttendee model was updated.
- Required mutations and queries were added to facilitate admin actions: inviting, registering, and checking-in users.
- Invited users can register for events.
- Refactor checkIn model 

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by aialok** [comment]
I will be fixing all the failed tests and please suggest me required changes if any.

**Comment by palisadoes** [comment]
*NOTE* Read very carefully

1. We just merged this PR which upgraded the prettier package. 
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/1840
1. It reformatted over 400 files. This will put your PR at risk of extensive merge conflicts.    
1. I suggest that before you merge your code to the latest upstream, or manually fix these merge conflicts that you:
    1. upgrade your `prettier` in your local branch to the same version
    1. run `prettier`
    
This will help to reduce the number of future merge conflicts for your PR.

**Comment by aialok** [comment]
@palisadoes, please review this pull request as we need a series of pull requests for this in talawa-admin and talawa-mobile. Also, if there is anything extra needed to be added, please do let me know.

File: schema.graphql
Comment (SiddheshKukade) [inline schema.graphql:1]: I wanted a clarification on why you've removed this fields

**Review by SiddheshKukade — CHANGES_REQUESTED** [review]
Kindly check the comments. Other than that PR looks good to merge. I think the fields that you've deleted may cause other apis to fail

**Comment by SiddheshKukade** [comment]
@aialok please take a pull from origin develop

File: schema.graphql
Comment (aialok) [reply schema.graphql:1]: We have already discussed the removal of these fields, as they are already being tracked elsewhere also removed fields have not been used previously, so refactoring the code is preferable if they are not being used.

For further reference:
https://github.com/PalisadoesFoundation/talawa-api/issues/1644#issuecomment-1879984882

**Comment by aialok** [comment]
> Kindly check the comments. Other than that PR looks good to merge. I think the fields that you've deleted may cause other apis to fail

I will be fixing all the failing api's if any.

**Comment by aialok** [comment]
@palisadoes, may I work on its frontend part until the PR gets reviewed? If I need anything extra in the frontend, then I will make the necessary changes in the backend accordingly.

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
Go ahead. I've approved and merged this

**Pull request merged**